client/my-sites/domains/domain-management/transfer/transfer-out/unlocked.jsx

```diff
@@ -62,81 +99,193 @@ class Unlocked extends React.Component {
 						break;
 				}
 				notices.error( errorMessage );
-
-				this.setState( { submitting: false } );
-			} else if ( hasPrivacyProtection ) {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked and ' +
-					'Privacy Protection has been enabled.' ) );
 			} else {
-				notices.success( translate( 'We\'ve canceled your domain transfer. Your domain is now locked back.' ) );
+				// Success.
+				this.setStateIfMounted( { sent: false } );
+
+				let successMessage;
+				if ( enablePrivacy && lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer. Your domain is now re-locked and ' +
+						'Privacy Protection has been enabled.' );
+				} else if ( enablePrivacy ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						'Privacy Protection has been re-enabled.' );
+				} else if ( lockDomain ) {
+					successMessage = translate( 'We\'ve canceled your domain transfer and ' +
+						're-locked your domain.' );
+				} else {
+					successMessage = translate( 'We\'ve canceled your domain transfer. ' );
+				}
+
+				notices.success( successMessage );
 			}
 		} );
 	};
 
-	handleResendConfirmationCodeClick = () => {
-		this.setState( { submitting: true } );
+	isDomainAlwaysTransferrable() {
+		const {
+			domainLockingAvailable,
+			hasPrivacyProtection,
+		} = getSelectedDomain( this.props );
+		return ! domainLockingAvailable && ! hasPrivacyProtection;
+	}
+
+	renderCancelButton( domain ) {
+		const { pendingTransfer } = domain;
+
+		if ( ! pendingTransfer && this.isDomainAlwaysTransferrable() ) {
```

Comment:
Yeah. If there's a pending transfer, the guard will always be false, so we won't bail out early  and we will show the cancel button - always show the button if there's a pending transfer.

It the domain is transferrable, skip the button, but only if there's no pending transfer.

```
> [true, false].forEach(pendingTransfer => [true, false].forEach( isDomainAlwaysTransferrable => console.log( 'pendingTransfer:',pendingTransfer,', isDomainAlwaysTransferrable:',isDomainAlwaysTransferrable, 'bail?', ( ! pendingTransfer && isDomainAlwaysTransferrable ) ) ) )
pendingTransfer: true , isDomainAlwaysTransferrable: true bail? false
pendingTransfer: true , isDomainAlwaysTransferrable: false bail? false
pendingTransfer: false , isDomainAlwaysTransferrable: true bail? true
pendingTransfer: false , isDomainAlwaysTransferrable: false bail? false
```

Would it be clearer to add the extra negations? `if ( ! ( pendingTransfer || ! isDomainAlwaysTransferrable ) ) bail`